### PR TITLE
Support for multiple clients and other improvements in load test config generation.

### PR DESCRIPTION
**Pull request by paulosjca**
This change adds the following features and improvements to load test generation:

*   Multiple client instances are supported by setting the option `--instances_per_client`.
*   `python_asyncio` is supported as a separate language.
*   `php` is renamed to `php7` to match the scenario language names.
*   Client and server names are shortened by the introduction of an index (`0`, `1`, ... with an optional prefix). This index replaces the uuid added to pod names when no component name is specified.
*   Driver pool and image substitution are added to template generation.
*   Empty entries are removed from the final configuration.

NOTE: Controller is currently not respecting driver settings from configuration, so driver is still named with a uuid. This will be addressed separately.

### CODE REVIEW

**Review by jtattermusch — APPROVED** [review]
Overall this is looking good but I didn't review in detail. 
I will let others perform the detailed review.

Thanks for bringing the support for multiple clients quickly.

**Comment by jtattermusch** [comment]
@drfloob  once this is in place, you could use the multiple client feature to experiment with the suspected degradation of C++ performance when there's multiple clients (this is tracked as https://github.com/grpc/grpc/issues/14899).
I'd be curios to see if the introduction of the new iomgr you're working on has any effect on this behavior.

**Comment by paulosjca** [comment]
> Overall this is looking good but I didn't review in detail.
> I will let others perform the detailed review.

There are not many people who can review this, so I will be merging it later today unless there are comments on it.

File: tools/run_tests/performance/README.md [hunk 1]
```diff
@@ -340,20 +348,38 @@ was generated from the example configurations in
...
     --name basic_all_languages
 ```
 
+The example template with prebuilt images in
+[loadtest_template_prebuilt_all_languages.yaml](./templates/loadtest_template_prebuilt_all_languages.yaml)
+was generated by the following command:
+
+```
+$ ./tools/run_tests/performance/loadtest_template.py \
+    -i ../test-infra/config/samples/*_example_loadtest_with_pre_built_workers.yaml \
```
Comment (drfloob) [inline tools/run_tests/performance/README.md:364]: nit: suggestion for consistency `-i ./config/samples/*_example_loadtest_with_pre_built_workers.yaml \`

**Review by drfloob — APPROVED** [review]
Looks good! I have a few use cases that would benefit from `instances_per_client`, I'll give it a shot. My scenarios are pretty simple, I've found it straightforward to manage static templates for my tasks (It took ~5 minutes to write the four of them I needed), so I imagine this will be more useful in more complicated scenarios.

File: tools/run_tests/performance/README.md (hunk 1 above)
Comment (paulosjca) [reply tools/run_tests/performance/README.md:364]: The input for the templates is coming from the test-infra repository. Later on we use the template (now in the grpc repo) to generate configs. The  paths in this example and in the loadtst_config.py example are different for this reason.

Comment (drfloob) [reply tools/run_tests/performance/README.md:364]: I see, thanks for explaining.

**Pull request merged**